GPU/Vulkan/TextureCacheVulkan.cpp

```diff
@@ -595,7 +648,21 @@ void TextureCacheVulkan::BuildTexture(TexCacheEntry *const entry, bool replaceIm
 			break;
 		}
 
-		bool allocSuccess = image->CreateDirect(cmdInit, w * scaleFactor, h * scaleFactor, maxLevel + 1, actualFmt, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL, VK_IMAGE_USAGE_TRANSFER_DST_BIT | VK_IMAGE_USAGE_SAMPLED_BIT, mapping);
+		VkImageLayout imageLayout = VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL;
+		VkImageUsageFlags usage = VK_IMAGE_USAGE_TRANSFER_DST_BIT | VK_IMAGE_USAGE_SAMPLED_BIT;
+		// If we want to use the GE debugger, we should add VK_IMAGE_USAGE_TRANSFER_SRC_BIT too...
```

Comment:
We can call `host->GPUDebuggingActive()` to determine if it's open.

-[Unknown]

Reply:
Though textures could be created before it was opened..

Reply:
Maybe we flush the texture cache when it opens?  Not ideal but it isn't completely terrible.

-[Unknown]

Reply:
Sure. Or we just always set the flag here - I can't imagine there's much of a cost to it really.

Reply:
At least for now, it seems like the debugger still works.

-[Unknown]

Reply:
It does work but with undefined behavior - so results will vary depending on GPU. In practice most layout transitions are no-ops on NVIDIA, some are on AMD and Intel too.